Hyperion: Fix fatal error in last merge

starkillerOG opened the pull request:
## Description:
In the last commit the self._brightness has been removed and named brightness.
However if the brightness is not in the current command this breaks the turn_on action because it is required for the cal_rgb.
This fixes that by using the self._brightness if it is not in the current command.

My last commit is alreay merged in this pull request:
https://github.com/home-assistant/home-assistant/pull/11543

## Checklist:
  - [x] The code change is tested and works locally.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support

- added brightness support to dim the hyperion light
- changed the "OFF" command to set the color to [0,0,0] after clearing all priorities.
  This is neccesary to keep the light turned off when an HDMI grabber is used for ambilight with hyperion.
- added HDMI ambilight mode recognition and control.
  by setting the "hdmi_priority" in your "configuration.yaml" file (defaults to 880), home assistant will now be able to recognize when the hyperion light is in HDMI ambilight mode and will change its icon to an HDMI symbol and set the status to ON.
Switching the hyperion light to HDMI ambilight mode can be done through the effect option (clears all priorities such that the HDMI grabber remains).
- added effect support for the default effects of hyperion, a custom list can be defined in the "configuration.yaml" file by using the "effect_list" option.

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support

- added brightness support to dim the hyperion light
- changed the "OFF" command to set the color to [0,0,0] after clearing all priorities.
  This is neccesary to keep the light turned off when an HDMI grabber is used for ambilight with hyperion.
- added HDMI ambilight mode recognition and control.
  by setting the "hdmi_priority" in your "configuration.yaml" file (defaults to 880), home assistant will now be able to recognize when the hyperion light is in HDMI ambilight mode and will change its icon to an HDMI symbol and set the status to ON.
Switching the hyperion light to HDMI ambilight mode can be done through the effect option (clears all priorities such that the HDMI grabber remains).
- added effect support for the default effects of hyperion, a custom list can be defined in the "configuration.yaml" file by using the "effect_list" option.
- fixed some style issues with too long lines

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support 

 - fixed some more indentation style issues

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support 

- yet more fixed visuel indent issues

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support 

- more visuel indents

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support

- fixed invalid variable "A"

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support

- remove unnececary brackets
- specify specific exceptions

starkillerOG committed:
correct changing state holding attributes during a service method

Proccesed the comments of @MartinHjelmare: https://github.com/home-assistant/home-assistant/pull/11543#pullrequestreview-88328659

starkillerOG committed:
indent correction

corrected tab instead of 4 spaces

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support 

- changed 'none' to None
- renamed "self._skip_check" to "self._skip_update"

starkillerOG committed:
Add brightness, HDMI and effect support 

changed checking if a list is empty from "list == []" to "not list"

starkillerOG committed:
Hyperion: Fix fatal error in last merge

In the last commit the self._brightness has been removed and named brightness.
However if the brightness is not in the current command this breaks the turn_on action because it is required for the cal_rgb.
This fixes that by using the self._brightness if it is not in the current command.

starkillerOG committed:
Merge branch 'dev' into patch-1

starkillerOG commented:
@MartinHjelmare could you merge this fix, since you also merged my original code (containing the mistake) in pull request https://github.com/home-assistant/home-assistant/pull/11543

MartinHjelmare reviewed (approved):
Thanks!

MartinHjelmare commented:
Can you rebase and clean up your branch so that you only have the latest commit on top of freshly synced dev?

starkillerOG commented:
@MartinHjelmare how do I sync the dev branch with my fork, I am new to github.

starkillerOG committed:
Merge pull request #1 from home-assistant/dev

Update my own fork

starkillerOG committed:
Hyperion: Fix fatal error in last merge

In the last commit the self._brightness has been removed and named brightness.
However if the brightness is not in the current command this breaks the turn_on action because it is required for the cal_rgb.
This fixes that by using the self._brightness if it is not in the current command.

My last commit is alreay merged in this pull request:
#11543

starkillerOG committed:
Merge pull request #2 from starkillerOG/HyperionFix

update own fork

starkillerOG commented:
I will just make a new pull request, I am messing this one up

starkillerOG commented:
new pull request: #11777
